### PR TITLE
feat: implementing language

### PR DESCRIPTION
**Pull request by MarLaslo**
This pull request introduces a new feature for managing languages in the CitMovie API, including the addition of a new controller, manager, repository, and associated DTOs. It also includes some minor changes and bug fixes in existing domain objects.

### New Feature: Language Management

* [`src/CitMovie.Api/Controller/LanguageController.cs`](diffhunk://#diff-9dd688c19645c6885d9562df6ef1845ad6b8a069e8981893b8bbb83e8db85c3cR1-R65): Added a new `LanguageController` to handle language-related endpoints, including methods for fetching languages with pagination.
* [`src/CitMovie.Business/Managers/ILanguageManager.cs`](diffhunk://#diff-7bfec1c23aa5abfc952860662749f2d7cc2b0870bb6cebc870e2d23c6ed89732R1-R10): Introduced the `ILanguageManager` interface for managing language-related business logic.
* [`src/CitMovie.Business/Managers/LanguageManager.cs`](diffhunk://#diff-817be7e2bff9b78bf87592c285b4afd72ca6dcfc8d133b197d67df654ac62ba4R1-R31): Implemented the `LanguageManager` class to handle business logic for languages, including fetching languages and their total count.
* [`src/CitMovie.Data/Repositories/ILanguageRepository.cs`](diffhunk://#diff-736a162eaca6d95fe28331d3c935e394b4c366f53e1a7e0f1cd4d43b5070ee1bR1-R13): Added the `ILanguageRepository` interface for language-related data access operations.
* [`src/CitMovie.Data/Repositories/LanguageRepository.cs`](diffhunk://#diff-c2ecd14cea1a0058e5a958dd1d4e77cd63301ed1407498bb20975bbb9cac4742R1-R49): Implemented the `LanguageRepository` class to interact with the database for language-related operations.
* [`src/CitMovie.Models/DTOs/LanguageDetailsDto.cs`](diffhunk://#diff-7d9bcf30756fbedc36cd79f98fdd358ebe83e4b38984cd546ff94f68208661b2R1-R8): Added the `LanguageDetailsDto` class to represent language details in data transfer objects.

### Dependency Injection

* [`src/CitMovie.Api/Program.cs`](diffhunk://#diff-751698e746b3ca3ade31345a336f4b2b41cb669558480af3cc413b5e185ec41dR19-L23): Registered `ILanguageRepository` and `LanguageManager` with the dependency injection container.

### Bug Fixes and Minor Changes

* [`src/CitMovie.Models/DomainObjects/Episode.cs`](diffhunk://#diff-bbd3a9d8b9e6fa7d4f9a66c2716d2986ab8135e255126bca8c15640717e18512R13): Added `[NotMapped]` attribute to the `Media` property to avoid mapping issues.
* [`src/CitMovie.Models/DomainObjects/Media.cs`](diffhunk://#diff-a25747ac88041a91062218f08926c6756071a8957c3b29ddacde8073c64dee9cR29-R35): Added `[NotMapped]` attribute to the `Season` and `RelatedMedia` properties to avoid mapping issues.
* [`src/CitMovie.Models/DomainObjects/Score.cs`](diffhunk://#diff-d9838b063396e8373bbbe0b5c12d36e961e4006bb31655c06913da9d732b5fc4L17-R17): Corrected the data type of the `MediaId` property from `string` to `int`.

These changes collectively enhance the CitMovie API by adding robust language management capabilities and addressing some minor issues in the domain models.

### CODE REVIEW

File: src/CitMovie.Api/Program.cs [hunk 1]
```diff
@@ -50,11 +50,11 @@
 builder.Services.AddControllers();
 builder.Services.AddScoped<IFollowRepository, FollowRepository>();
 builder.Services.AddScoped<FollowService>();
+builder.Services.AddScoped<ILanguageRepository, LanguageRepository>();
+builder.Services.AddScoped<LanguageManager>();
```
Comment (sofushn) [inline src/CitMovie.Api/Program.cs:54]: Please use the `CitMovieServiceCollectionExtensionsAddCitMovieServices()` method in the business layer to register dependencies

File: src/CitMovie.Data/Repositories/ILanguageRepository.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace CitMovie.Data;
+
+public interface ILanguageRepository
+{
+    Task<Language?> GetLanguageByIdAsync(int languageId);
+    Task<IList<Language>> GetLanguagesAsync( int page, int pageSize);
+    Task <Language> CreateLanguageAsync(Language language);
+    Task <Language> UpdateLanguageAsync(Language language);
+    Task <Language>DeleteLanguageAsync(int languageId);
```
Comment (sofushn) [inline src/CitMovie.Data/Repositories/ILanguageRepository.cs:11]: Should we just delete these if we don't use them in the code?

File: src/CitMovie.Data/Repositories/LanguageRepository.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+        public Task<Language> DeleteLanguageAsync(int languageId)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task<int> GetTotalFollowingsCountAsync()
+        {
+            return await _context.Languages.CountAsync();
+        }
```
Comment (sofushn) [inline src/CitMovie.Data/Repositories/LanguageRepository.cs:47]: Should we just delete these if we don't use them in the code?

File: src/CitMovie.Data/DataContext.cs [hunk 4]
```diff
@@ -28,19 +31,17 @@ public class DataContext : DbContext
...
     
+    public DataContext(string connectionString) {
+        _connectionString = connectionString;
+    }
+    
+    [ActivatorUtilitiesConstructor]
     public DataContext(DbContextOptions<DataContext> options) 
         : base(options) { }
     protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder) {
+        var connectionString = "Port=5433;Database=portfolio_database;Username=postgres;Password=postgres";
```
Comment (sofushn) [inline src/CitMovie.Data/DataContext.cs:42]: I would guess this was only used for testing, so should be removed again

File: src/CitMovie.Data/DataContext.cs [hunk 5]
```diff
@@ -28,19 +31,17 @@ public class DataContext : DbContext
...
         if (!optionsBuilder.IsConfigured)
-        {
-            var configuration = new ConfigurationBuilder()
-                .SetBasePath(Directory.GetCurrentDirectory())
-                .AddJsonFile("appsettings.json")
-                .AddEnvironmentVariables()
-                .Build();
-
-            optionsBuilder.UseNpgsql(configuration.GetConnectionString("PostgresConnection"));
-        }
```
Comment (sofushn) [inline src/CitMovie.Data/DataContext.cs:43]: Nice that you removed this, I did the same in the framework context 👍

**Review by sofushn — CHANGES_REQUESTED** [review]

File: src/CitMovie.Api/Program.cs (hunk 1 above)
Comment (MarLaslo) [reply src/CitMovie.Api/Program.cs:54]: Nice I have not realized we have this :D was thinking we should do something like that

File: src/CitMovie.Data/Repositories/ILanguageRepository.cs (hunk 2 above)
Comment (MarLaslo) [reply src/CitMovie.Data/Repositories/ILanguageRepository.cs:11]: Sure we can

**Pull request merged**